### DISCUSSION TITLE
Enhancement suggestion for the HR thread

### DISCUSSION THREAD

**Discussion by howardlowndes** [ledgersmb/LedgerSMB#7879] [Ideas]
### Version

1.11.6

### What browsers are you seeing the problem on?

This problem isn't browser related

### What happened?

I mentioned a few weeks back that my interest in LedgerSMB is for a computer based book keeping system for our Albury Men;s Shed. We were the first manual activities centre ib Australia aimed toward providing an encouraging environment for retired men who had been used to a manual, often skilled, workplace environment, having been established 45 years ago, and now part of the Australian Men's Shed Association, itself established 30 years ago.
Our book keeping needs are simple but in some ways unique. We are a registered charity (with the Australian Tax Office and others) and are exempt from charging  Output Goods and Services Tax (GST) but can claim back Input GST that we are charged on our purchases. Also being an Incorporated Association there is a need for auditible book keeping, hence the look at LedgerSMB, as we now accept Debit/Credit cards as well as direct deposits.
Even more so is the need to look to our HR recording. We are all volunteers so there is no complications from Payroll, Payroll Tax, Withholding Tax, Superannuation, etc, but our members do work with some potentially dangerous machinery and therefore Workplace Health and Safety (WHS) considerations come into play and we have a requirement to maintain record of initlal and ongoing training and compliance with safe work practices. To this end the management committee would like to investigate the HR thread of the project as a means of recording and maintaining these vital records which could be a quite disparate range and would vary among other LedgerSMB users

### What should have happened?

For discussion, consideration, and planning.

**Comment by ehuelsmann** [comment]
Thanks for bringing this up! For Parts, we're looking to develop the ability to register "custom" attributes: attributes a business may want to define by themselves. Use-cases could be "ingredients lists", but could be anything else.

Would the ability to register your own attributes (e.g. an attribute per type of training followed initially and updated training) solve your use-case?

Thinking about the use-case a bit more: you probably want to keep a full log of all trainings followed of every type. In that case: would it be practical to use the Notes field? Maybe we'd need to make them searchable, but they will be able to keep the full log (notes are currently append-only).

**Comment by howardlowndes** [comment]
Certainly the ability to define our own attributes would have benefits.
For the training scenario that I have in mind we would need to hold
"Initial training", "Review due date" (refreshing each time), "Review
result" (also refreshing each time), and a "Notes" (probably additive)
record to supplement the latter. This would need to be across a range of
disparate  equipments.
The whole thing would need to be reasonably generic to suit a range of HR
needs.
Foes this provide some input?

On Fri, 29 Dec 2023 at 22:32, Erik Huelsmann ***@***.***>
wrote:

> Thanks for bringing this up! For Parts, we're looking to develop the
> ability to register "custom" attributes: attributes a business may want to
> define by themselves. Use-cases could be "ingredients lists", but could be
> anything else.
>
> Would the ability to register your own attributes (e.g. an attribute per
> type of training followed initially and updated training) solve your
> use-case?
>
> Thinking about the use-case a bit more: you probably want to keep a full
> log of all trainings followed of every type. In that case: would it be
> practical to use the Notes field? Maybe we'd need to make them searchable,
> but they will be able to keep the full log (notes are currently
> append-only).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ledgersmb/LedgerSMB/discussions/7879#discussioncomment-7971574>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMS2F7WY2252R6NQDKMMH3TYL2S4JAVCNFSM6AAAAABBGQDOI2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TSNZRGU3TI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Howard.
-- 
When you want a computer system that works, just choose Linux;
When you want a computer system that works, just, choose Microsoft.
